app/utility/base_planning_svc.py

```diff
@@ -142,3 +148,14 @@ def _is_fact_bound(fact):
             if not requirement.enforce(link.used, relationships):
                 return False
         return True
+
+    async def _exclude_existing(self, link, operation):
+        all_hostnames = [agent.host for agent in await operation._active_agents()]
+        for item in link.relationships:
+            # prevent backwards lateral movement
+            if 'remote.host' in item.trait:
+                if item.value.split('.')[0].lower() in all_hostnames:
+                    return False
+                elif any(h in item.value.split('.')[0].lower() for h in all_hostnames):
+                    return False
```

Comment:
Shouldn't this be a `if item.value.split('.')[0].lower() in all_hostnames or any(h in item.value.split('.')[0].lower() for h in all_hostnames):`